pulumi/volumeReaper/src/cleanup.ts

```diff
@@ -0,0 +1,74 @@
+import k8s, { KubernetesObject } from '@kubernetes/client-node'
+import { VolumeSnapshot } from './snapshot'
+
+export const cleanup = async (
+  k8sApi: k8s.KubernetesObjectApi,
+  sts: string,
+  namespace: string,
+  pvcsToKeepCount: number
+) => {
+  const snapshots = await getExistingVolumeSnapshots(k8sApi, sts, namespace)
+  if (snapshots.length > pvcsToKeepCount) {
+    await removeOldSnapshots(k8sApi, snapshots, pvcsToKeepCount)
+  }
+
+  console.log('Snapshot cleanup completed')
+}
+
+const getExistingVolumeSnapshots = async (
+  k8sApi: k8s.KubernetesObjectApi,
+  sts: string,
+  namespace: string
+): Promise<KubernetesObject[]> => {
+  const snapshots = await k8sApi.list<VolumeSnapshot>(
+    'snapshot.storage.k8s.io/v1',
+    'VolumeSnapshot',
+    namespace,
+    undefined,
+    undefined,
+    undefined,
+    undefined,
+    `statefulset=${sts}`
+  )
+  const items = snapshots.body.items
+  console.log(`Found ${items.length} snapshots for ${namespace}.${sts}`)
+  return items.sort(sortByCreationTimestampDesc)
+}
+
+const sortByCreationTimestampDesc = (a: VolumeSnapshot, b: VolumeSnapshot) =>
+  new Date(b.metadata?.creationTimestamp).getTime() - new Date(a.metadata?.creationTimestamp).getTime()
```

Comment:
same here, now that we are using `VolumeSnapshot`
```suggestion
  new Date(b.metadata.creationTimestamp).getTime() - new Date(a.metadata.creationTimestamp).getTime()
```